cesmcatalog/gen_CESM_catalog.py

```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/env python
+"""
+  Goal: a script that gets pointed to a CESM case directory and generates
+        an intake-esm catalog of output in the short-term archive directory
+
+  Current state: a script that gets pointed to a CESM case directory and prints
+                 some environment variables that will be necessary for above
+
+  NOTE: this script needs to be run in the CESM postprocessing python environment,
+        which is still based on python 2.7. Follow instructions at
+
+        https://github.com/NCAR/CESM_postprocessing/wiki/cheyenne-and-DAV-quick-start-guide
+"""
+
+import argparse
+import fnmatch
+import logging
+import os
+import subprocess
+import sys
+
+################################################################################
+
+
+def _parse_args():
+    """ Wrapper for argparse, returns dictionary of arguments """
+
+    parser = argparse.ArgumentParser(
+        description='Generate intake-esm catalog for a CESM case',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    parser.add_argument(
+        '-c',
+        '--case-root',
+        action='store',
+        dest='case_root',
+        required=True,
+        help='CESM case root to generate intake-esm catalog for',
+    )
+    parser.add_argument(
+        '-d',
+        '--debug',
+        action='store_true',
+        dest='debug',
+        help='Add additional output for debugging',
+    )
+
+    return parser.parse_args()
+
+
+################################################################################
+
+
+def _find_data(case_root):
+    logger = logging.getLogger(__name__)
+
+    # 1. change directory to case_root (if it exists)
+    try:
+        os.chdir(case_root)
+    except:
+        # TODO: set up logger instead of print statements
+        logger.error('{} does not exist'.format(case_root))
+        sys.exit(1)
+
+    # 2. Collect info on where time slice output is
+    if not os.path.isfile('./xmlquery'):
```

Comment:
Instead of using xmlquery I think that you should use cime objects directly, for example:

  with Case(first_case, read_only=False) as case:
       run_refcase = case.get_value('RUN_REFCASE')

Reply:
I'd be happy to use the `Case` class, but struggled to figure out how to access it. Before I can import `Case` I need to know `CIMEROOT` right? Or are the python libraries symlinked in case root?

```
sys.path.append(os.path.join(subprocess.check_output('./xmlquery --value CIMEROOT', shell=True), 'scripts', 'lib'))
from CIME.case import Case
```

works, but I assume there's a cleaner way?

Reply:
There are a number of ways to do it, the below uses an environment variable.  In cime we symlink the files in the case directory to their location in the cime source tree which also works (for example xmlquery). 

```
cimeroot = os.environ.get('CIMEROOT')
_LIBDIR = os.path.join(cimeroot, "scripts", "Tools")                                                                         
sys.path.append(_LIBDIR)                                                                                                             
                                                                                                                                     
from standard_script_setup          import *                                                                                         
                                                                                                                                     
from CIME.case import Case                                                                                                           
                                                                                                                                     
logger = logging.getLogger(__name__)                        
```

Reply:
I added an optional `--cimeroot` option to the script, and if it isn't available then I call `xmlquery` once to get it added to the path, but the rest is much cleaner:

```
    run_config = dict()
    with Case(caseroot, read_only=True) as case:
        for var in ['CASE', 'GET_REFCASE', 'RUN_REFCASE', 'RUN_REFDATE', 'RUN_STARTDATE']:
            run_config[var] = case.get_value(var)

        if case.get_value('DOUT_S') == 'TRUE':
            DOUT_S_ROOT = case.get_value('DOUT_S_ROOT')
        else:
            DOUT_S_ROOT = None
```

Once this script is called by CESM postprocessing, we'll either use `--cimeroot` or I'll modify it to check the environment variable. Thanks for the pointers here!